torch_dicom/transforms/overlay.py

```diff
@@ -0,0 +1,228 @@
+        target_for_watermark = torch.where(alpha > 0, watermark, target_for_watermark)
+        inpt[..., bounds[0] : bounds[2], bounds[1] : bounds[3]] = target_for_watermark
+        return inpt
+
+    def _get_params(self, flat_inputs: List[Any]) -> Dict[str, Any]:
+        img = next(iter(i for i in flat_inputs if isinstance(i, TVImage)))
+        H, W = img.shape[-2:]
+        watermark_idx = int(torch.randint(len(self.paths), ()))
+        augment = bool(torch.rand(1) < self.p)
+        scale = float(torch.rand(1) * (self.scale_range[1] - self.scale_range[0]) + self.scale_range[0])
+        center = torch.rand(2, device=img.device) * torch.tensor([H, W], device=img.device)
+        center = center.round().long()
+        return {
+            "augment": augment,
+            "watermark_idx": watermark_idx,
+            "scale": scale,
+            "center": center,
+        }
+
+    def _load_image(self, idx: int) -> Tensor:
+        """
+        Load an image from the list of paths based on the provided index.
+
+        Args:
+            idx: The index of the image to load.
+
+        Returns:
+            The loaded image as a Tensor.
+        """
+        if not 0 <= idx < len(self.paths):
+            raise IndexError(f"Index {idx} out of range")  # pragma: no cover
+        path = self.paths[idx]
+
+        img = Image.open(path)
+        if img.mode != "LA":
+            img = img.convert("LA")
+
+        img = ToImage()(img)
+        img = ConvertImageDtype(torch.float32)(img)
+        return img
+
+
+class AddWatermark(AddOverlay):
+    r"""Add a machine watermark to the image. This transform will attempt to place the watermark
+    in an empty region of the image. Currently, only `.png` images with a luminance and alpha channel
+    are supported.
+
+    Args:
+        path: Path to directory containing overlay `.png` images.
+        p: Probability of applying the transform.
+        scale: Range of scales to apply to the overlay.
+        force_in_bounds: If `True`, the overlay will be translated to ensure it is fully contained within the image.
+        grid_size: Size of the grid to use when searching for an empty region of the image.
+        use_bar: If `True`, a progress bar will be displayed when loading the watermark images.
+    """
+
+    def __init__(
+        self,
+        path: Path,
+        p: float = 0.2,
+        scale: Tuple[float, float] = (0.75, 1.25),
+        force_in_bounds: bool = True,
+        grid_size: int = 4,
+        use_bar: bool = True,
+    ):
+        super().__init__(path, p, scale, force_in_bounds, use_bar)
+        self.grid_size = grid_size
+
+    def _get_params(self, flat_inputs: List[Any]) -> Dict[str, Any]:
+        img = next(iter(i for i in flat_inputs if isinstance(i, TVImage)))
+        params = super()._get_params(flat_inputs)
+
+        # Select the grid cell where the watermark will be placed
+        quadrant = self._select_grid_cell(img, grid_size=self.grid_size)
```

Comment:
Since this is only a quadrant in the case that `grid_size == 2`, maybe we should just call this `grid_cell` (or similar)?